src/redux/checkout/tests/checkout.actions.test.js

```diff
@@ -241,7 +257,356 @@ describe('get prices from NovaPoshta API', () => {
     expect(getNovaPoshtaPrices(payload)).toBeTruthy();
   });
 
-  it('value by key "type" of returned object should to be equal to "GET_NOVAPOSHTA_PRICES"', () => {
+  it('value by key "type" of returned object should be equal to "GET_NOVAPOSHTA_PRICES"', () => {
     expect(getNovaPoshtaPrices(payload).type).toEqual(type);
   });
 });
+
+// regions from UkrPoshta action tests
+describe('set regions from UkrPoshta action test', () => {
+  let type;
+  let setUPRegionsAction;
+
+  const payload = [
+    {
+      REGION_ID: '1',
+      REGION_UA: 'Вінницька',
+      REGION_EN: 'Vinnytska'
+    },
+    {
+      REGION_ID: '2',
+      REGION_UA: 'Волинська',
+      REGION_EN: 'Volynska'
+    }
+  ];
+
+  beforeEach(() => {
+    type = SET_UKRPOST_REGIONS;
+    setUPRegionsAction = setUkrPostRegions(payload);
+  });
+
+  it('should return object', () => {
+    expect(typeof setUPRegionsAction).toStrictEqual('object');
+  });
+
+  it('should be truthy', () => {
+    expect(setUPRegionsAction).toBeTruthy();
+  });
+
+  it('value by key "type" of returned object should be equal to "SET_UKRPOST_REGIONS"', () => {
+    expect(setUPRegionsAction.type).toEqual(type);
+  });
+
+  it('value by key "payload[0].REGION_UA" of returned object should be equal to "Вінницька"', () => {
+    expect(payload[0].REGION_UA).toEqual('Вінницька');
+  });
+
+  it('value by key "payload[0].REGION_ID" of returned object should to be "1"', () => {
+    expect(payload[0].REGION_ID).toEqual('1');
+  });
+});
+
+describe('get regions from UkrPoshta API', () => {
+  let type;
+
+  beforeEach(() => {
+    type = GET_UKRPOST_REGIONS;
+  });
+
+  it('should return object', () => {
+    expect(typeof getUkrPostRegions()).toStrictEqual('object');
+  });
+
+  it('should to be truthy', () => {
+    expect(getUkrPostRegions()).toBeTruthy();
+  });
+
+  it('value by key "type" of returned object should to be equal to "GET_NOVAPOSHTA_WAREHOUSES"', () => {
+    expect(getUkrPostRegions().type).toEqual(type);
+  });
+});
+
+// districts from UkrPoshta action tests
+describe('set districts from UkrPoshta action test', () => {
+  let type;
+  let setUPDistrictsAction;
+
+  const payload = [
+    {
+      REGION_ID: '4',
```

Comment:
remove magic strings, get this array from variables

Reply:
done